### PR TITLE
[DISCO-1787] Migrate molotov-based load tests to Locust

### PR DESCRIPTION
**Pull request by Trinaa**
## References

JIRA: [DISCO-1787](https://mozilla-hub.atlassian.net/browse/DISCO-1787)
GitHub: #390 

## Description

- Add a Locust load testing solution
    - This load testing solution was migrated from the [Molotov based solution](https://github.com/mozilla-services/contile-loadtests) and based off the [Merion load testing solution](https://github.com/mozilla-services/merino-py/tree/main/tests/load)
    - The Locust load test solution was executed on April 23rd 2023, [results](https://docs.google.com/spreadsheets/d/1lGHu--eXEy6ShErmU1-SQ26yLY4xOjGubnRXt0DdGB0/edit#gid=1576418580).

- The statistics are presented using the format {endpoint} - {user-agent heading} - {location heading}, but could be presented more concisely, suggestions for reporting format are welcome. Examples:
    - Granular Statistics: image
    - Concise Statistics: image

[DISCO-1787]: https://mozilla-hub.atlassian.net/browse/DISCO-1787?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test-engineering/load/locustfiles/load.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            ShapeStage(
+                run_time=(self.RUN_TIME - 60),
+                users=self.USERS,
+                spawn_rate=spawn_rate,
+            ),
+            # Stage 2: Stop users in the last minute
+            ShapeStage(run_time=self.RUN_TIME, users=0, spawn_rate=spawn_rate),
+        ]
+
+    def tick(self) -> tuple[int, float, list[Type[User]] | None] | None:
```
Comment (ncloudioj) [inline test-engineering/load/locustfiles/load.py:46]: Nit: let's use a type alias to make this easier to read.

```
TickReturnType = tuple[int, float, list[Type[User]] | None]
def tick(self) -> TickReturnType | None:
    ...
```

**Review by ncloudioj — APPROVED** [review]
This looks great, thanks @Trinaa!

File: test-engineering/load/locustfiles/load.py (hunk 1 above)
Comment (Trinaa) [reply test-engineering/load/locustfiles/load.py]: Change applied: https://github.com/mozilla-services/contile/pull/536/commits/8e6ca2b4266b6d43fe0ba380a9b63bbe9123f1b1

File: .circleci/config.yml [hunk 2]
```diff
@@ -280,6 +280,31 @@ jobs:
...
+    working_directory: "~/contile/test-engineering/load"
+    steps:
+      - checkout:
+          path: ~/contile/
+      - run:
+          name: setup poetry
+          command: |
+            poetry -V
+            poetry install
+      - run:
```
[the remark this answers is not in the document]
Comment (ncloudioj) [reply .circleci/config.yml:295]: sounds good.

**Review by ncloudioj — APPROVED** [review]

**Comment by stalkerg** [comment]
Migration from async/await to gevent it's sound like huge degradation.

**Pull request merged**